include/exadg/operators/operator_base.cpp

```diff
@@ -96,20 +96,8 @@ OperatorBase<dim, Number, n_components>::reinit(MatrixFree<dim, Number> const &
   // initialize n_mpi_proceses
   DoFHandler<dim> const & dof_handler = this->matrix_free->get_dof_handler(this->data.dof_index);
 
-  MPI_Comm comm;
-
-  // extract communicator
-  {
-    auto tria =
-      dynamic_cast<parallel::TriangulationBase<dim> const *>(&dof_handler.get_triangulation());
-
-    if(tria != NULL)
-      comm = tria->get_communicator();
-    else // not distributed triangulation
-      comm = MPI_COMM_SELF;
-  }
-
-  n_mpi_processes = Utilities::MPI::n_mpi_processes(comm);
+  Triangulation<dim> const & tria = dof_handler.get_triangulation();
+  n_mpi_processes                 = Utilities::MPI::n_mpi_processes(tria.get_communicator());
```

Comment:
Should we simplify this to
```suggestion
  n_mpi_processes                 = Utilities::MPI::n_mpi_processes(dof_handler.get_triangulation().get_communicator());
```
or, even simpler,
```suggestion
  n_mpi_processes                 = Utilities::MPI::n_mpi_processes(dof_handler.get_communicator());
```